src/plugins/discover/public/components/doc_table/doc_table_wrapper.tsx

```diff
@@ -72,7 +72,7 @@ export interface DocTableProps {
   /**
    * Filter callback
    */
-  onFilter: DocViewFilterFn;
+  onFilter?: DocViewFilterFn;
```

Comment:
What makes this prop optional for the legacy table?

Reply:
It is a mess here, it is optional in other places and required to others. I dont allow the onFilter on the embeddable so this means that it should be optional. This change is fixing this